openssl/ecdh.go

```diff
@@ -16,10 +16,22 @@ import (
 type PublicKeyECDH struct {
 	_pkey C.GO_EVP_PKEY_PTR
 	bytes []byte
+
+	// priv is only set when _pkey is derived from a private key,
+	// in which case the private key is responsible of freeing _pkey.
+	// This also ensures that priv is not finalized meanwhile
+	// the public key is alive.
```

Comment:
```suggestion
	// priv is only set when PublicKeyECDH is derived from a private key,
	// in which case priv's finalizer is responsible for freeing _pkey.
	// This ensures priv is not finalized while the public key is alive,
	// which could cause use-after-free and double-free behavior.
```

When this struct comes from `PrivateKeyECDH.PublicKey`, `_pkey` isn't really "derived" from it, it's just copied over, right? The `bytes` being the thing that's derived. (Suggestion is to just say the whole struct is derived, which I think makes sense.)